crates/builder/src/tasks/submit.rs

```diff
@@ -156,6 +166,10 @@ impl SubmitTask {
                 "error in transaction submission"
             );
 
+            if e.message.contains(&hex::encode(IncorrectHostBlock::SELECTOR)) {
```

Comment:
use this:

https://github.com/alloy-rs/alloy/blob/3fc5943bd84c3d9883dc06118a0367a084668bdf/crates/json-rpc/src/response/error.rs#L343